python/pyspark/sql/session.py

```diff
@@ -2002,6 +2003,61 @@ def addArtifacts(
 
     addArtifact = addArtifacts
 
+    def registerProgressHandler(self, handler: "ProgressHandler") -> None:
+        """
+        Register a progress handler to be called when a progress update is received from the server.
+
+        .. versionadded:: 4.0
+
+        Parameters
+        ----------
+        handler : ProgressHandler
+          A callable that follows the ProgressHandler interface. This handler will be called
+          on every progress update.
+
+        Examples
+        --------
+
+        >>> def progress_handler(stages, inflight_tasks, done):
+        ...     print(f"{len(stages)} Stages known, Done: {done}")
+        >>> spark.registerProgressHandler(progress_handler)
+        >>> res = spark.range(10).repartition(1).collect()
+        3 Stages known, Done: False
+        3 Stages known, Done: True
```

Comment:
This test is flaky:

```
File "/__w/spark/spark/python/pyspark/sql/connect/session.py", line 346, in pyspark.sql.connect.session.SparkSession.registerProgressHandler
Failed example:
    res = spark.range(10).repartition(1).collect()
Expected:
    3 Stages known, Done: False
    3 Stages known, Done: True
Got:
    0 Stages known, Done: True
```

https://github.com/apache/spark/actions/runs/8564043093/job/23470007059.

Let me skip it for now.

Reply:
I'll unflake it. Thanks!